scripts/convert-lockfile.lua

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env -S nvim -l
+vim.opt.runtimepath:append('.')
+local util = require('nvim-treesitter.util')
+local parsers = require('nvim-treesitter.parsers')
+
+local filename = require('nvim-treesitter.install').get_package_path('lockfile.json')
+local lockfile = vim.json.decode(util.read_file(filename)) --[[@as table<string,{revision:string}>]]
+
+for k, p in pairs(parsers) do
+  if p.install_info then
+    p.install_info.revision = lockfile[k].revision
+  end
+end
+
+-- write new parser file
+local header = '---@type nvim-ts.parsers\nreturn '
+local parser_file = header .. vim.inspect(parsers)
+if vim.fn.executable('stylua') == 1 then
+  parser_file = vim.system({ 'stylua', '-' }, { stdin = parser_file }):wait().stdout --[[@as string]]
```

Comment:
Do we need `@as string`? `stdout` should already be a string? If it's a `string?` then I find it's better to wrap in `assert`.

Reply:
yep, it's about the  `string?`. `assert` is fine; or just live with the warning (it's a throw-away script to help with rebasing and not meant to be permanent).